testing/src/test/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java

```diff
@@ -259,6 +257,26 @@ public void testInvalidKeyspaceAndTable() throws IOException {
                 keyspace));
   }
 
+  @Test
+  public void testDocGetOnNotExistingNamespace() throws IOException {
+    String resp =
+        RestUtils.get(
+            authToken, hostWithPort + "/v2/namespaces/unknown/collections/collection/1", 404);
+    assertThat(resp)
+        .isEqualTo(
+            "{\"description\":\"Not found: Namespace unknown does not exist.\",\"code\":404}");
```

Comment:
```suggestion
    Error response = objectMapper.readValue(resp, Error.class);
    assertThat(response.getCode()).isEqualTo(HttpStatus.SC_NOT_FOUND);
    assertThat(response.getDescription()).isEqualTo("Not found: Namespace unknown does not exist.");
```

Reply:
I would rather not do it this way and this also stands for other comments/suggestions you provided, let me explain why. So these are integration tests. From the outside there is no object mapper, the request body and responses are strings. It's very easy to have some really nice bugs hidden if you use your internal object mapper to actually produce request bodies (trust me I have been there). 

Furthermore, wrt responses I would rather keep this as string, imagine a situation where you add a new field to the `Error` class. The changes above will keep the test green, although you actually changed the response sent back. In such situation, I would like a test to fail, so that we can be informed and we can say yep this was intentional, we extended the error class and let me fix the test.

Does this make sense?

Reply:
>It's very easy to have some really nice bugs hidden if you use your internal object mapper to actually produce request bodies (trust me I have been there).

You have a point. I have seen this myself. 

As for the the use of Error.class this is the way we have been doing (see here: https://github.com/stargate/stargate/blob/master/testing/src/test/java/io/stargate/it/http/RestApiv2Test.java#L1494). So, for the sake of consistency it would be nice to adhere to that or refactor those other uses. If we change the Error class with a new field the tests will break if this new field is required. OTOH, if it's optional so it should not mess with other tests besides the tests that actually need this new field, ideally.

Reply:
So you mean refactor all uses in the `BaseDocumentApiV2Test.java ` to this what you liked above? Cause in this test class it has been direct string assertion. Actually if I am up to refactor stuff, seems that some kind of json path assert is way better imo, what do you think about that?

Reply:
Refactoring to use JSON asserts is a good idea, I think. As far as this PR goes, I think string asserts are ok for the new functionality. I'd be +1 to do the refactoring in a dedicated (separate) PR.

Reply:
I would agree with @dimas-b, let's then merge this I guess..